components/public-api/gitpod/v1/scm.proto

```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service SCMService {
+  rpc GetToken(GetTokenRequest) returns (GetTokenResponse) {}
+  rpc GuessTokenScopes(GuessTokenScopesRequest) returns (GuessTokenScopesResponse) {}
+  rpc SearchRepositories(SearchRepositoriesRequest) returns (SearchRepositoriesResponse) {}
```

Comment:
It should be ListRepositories

Reply:
@akosyakov, could you explain why a search should be treated as a listing?

Reply:
because List is search in all APIs for given resource

Reply:
https://cloud.google.com/apis/design/standard_methods#list 

> List is commonly used to search for resources. 

We shoudl not introduce non standard methods without very good justificaiton, i.e. if it is very prominient operation which will do some kind of side effect besides of search.

Reply:
there is though suggestion for common non standard methods, https://cloud.google.com/apis/design/custom_methods#common_custom_methods

with search if it does not confrom to stadard semantics

Reply:
Do you mean it is because we don't list by ids, but always query by string?

Reply:
@akosyakov there is a significant difference between search and list, and here we rely on search results of 3rd party services. How can we call it listing, if we in fact use search API under the hood? We cannot uphold the meaning of listing. Search may include new items in results with a delay after its creation in a system. It all depends on load and availability.

Reply:
> Search > Alternative to List for fetching data that does not adhere to List semantics, such as services.search.

https://cloud.google.com/apis/design/custom_methods#common_custom_methods

> with search if it does not confrom to stadard LIST semantics

It doesn't.